utils/schemed-kafka-components-utils/src/test/scala/pl/touk/nussknacker/engine/schemedkafka/encode/BestEffortJsonSchemaEncoderTest.scala

```diff
@@ -0,0 +1,117 @@
+
+class BestEffortJsonSchemaEncoderTest extends AnyFunSuite {
+
+  val encoder = new BestEffortJsonSchemaEncoder(ValidationMode.strict)
+
+  test("should encode object") {
+    val schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$schema": "https://json-schema.org/draft-07/schema",
+        |  "type": "object",
+        |  "properties": {
+        |    "firstName": {
+        |      "type": "string"
+        |    },
+        |    "lastName": {
+        |      "type": "string"
+        |    },
+        |    "age": {
+        |      "type": "integer"
+        |    }
+        |  }
+        |}""".stripMargin))
+
+
+    val encoded = encoder.encode(Map(
+      "firstName" -> "John",
+      "lastName" -> "Smith",
+      "age" -> 1L
+    ), schema)
+
+    encoded shouldEqual Valid(Json.obj(
+      "firstName" -> Json.fromString("John"),
+      "lastName" -> Json.fromString("Smith"),
+      "age" -> Json.fromLong(1),
+    ))
+  }
+
+  test("should encode string") {
+    val schema: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$schema": "https://json-schema.org/draft-07/schema",
+        |  "type": "string"
+        |}""".stripMargin))
+
+    val encoded = encoder.encode("1", schema)
+
+    encoded shouldEqual Valid(Json.fromString("1"))
+  }
+
+  test("should encode number") {
+    val schema: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$schema": "https://json-schema.org/draft-07/schema",
+        |  "type": "number"
+        |}""".stripMargin))
+
+    val encoded = encoder.encode(1L, schema)
+
+    encoded shouldEqual Valid(Json.fromLong(1L))
+  }
+
+  test("should encode array") {
+    val schema: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$schema": "https://json-schema.org/draft-07/schema",
+        |  "type": "array",
+        |  "items": {
+        |    "type": "number"
+        |  }
+        |}""".stripMargin))
+
+    val encoded = encoder.encode(List(1), schema)
+
+    encoded shouldEqual Valid(Json.arr(Json.fromLong(1L)))
+  }
+
+  test("should throw when value and schema type mismatch") {
+    val schema: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$schema": "https://json-schema.org/draft-07/schema",
+        |  "type": "number"
+        |}""".stripMargin))
+
+    val encoded = encoder.encode("1", schema)
+
+    encoded shouldEqual Invalid(NonEmptyList("Not expected type: java.lang.String for field: None with schema: {\"type\":\"number\",\"$schema\":\"https://json-schema.org/draft-07/schema\"}", List()))
+  }
+
+  test("should accept redundant parameters if validation modes allows this") {
+    val schema: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$schema": "https://json-schema.org/draft-07/schema",
+        |  "type": "object",
+        |  "properties": {
+        |    "foo": {
+        |      "type": "string"
+        |    }
+        |  }
+        |}""".stripMargin))
```

Comment:
can we also have tests for cases when we try to encode:
 * Map()
 * Map("foo", null)
against such schema, and also when `foo` is required in schema